### ISSUE TITLE
Roc hangs when ingesting bytes of large file

### ISSUE THREAD

**Issue by lukewilliamboswell** [roc-lang/roc#5729]
Using this [twitter.json](https://github.com/miloyip/nativejson-benchmark/blob/master/data/twitter.json) file which is 662kb on disk seems to cause roc to hang when ingesting the bytes directly. Below is a test app which counts the number of slashes, however it seems to hang when building if the bytes are ingested directly. 

```
% roc run Test.roc <-- this is opening the file then counting backslashes
Counted 1230 backslashes
% roc run Test.roc <-- this is ingesting the file bytes into the Roc app module
^C
```

```elixir
app "broken"
    packages {
        pf: "https://github.com/roc-lang/basic-cli/releases/download/0.4.0/DI4lqn7LIZs8ZrCDUgLK-tHHpQmxGF1ZrlevRKq5LXk.tar.br",
    }
    imports [
        pf.Stdout,
        # pf.Path,  
        # pf.File,
        # pf.Task,
        "twitter.json" as fileContents : List U8,
    ]
    provides [main] to pf

# count the number of backslashes in the input file
main =
    # fileName = "twitter.json"
    # path = Path.fromStr fileName
    # fileContents <- 
    #     File.readUtf8 path 
    #     |> Task.onFail \_ -> crash "Failed to read file"
    #     |> Task.map Str.toUtf8
    #     |> Task.await
    
    count = 
        fileContents
        |> List.map \a -> if a == '\\' then 1 else 0
        |> List.sum 

    Stdout.line "Counted \(Num.toStr count) backslashes"
```

**Comment by lukewilliamboswell** [comment]
@bhansconnect you may be interested in this

**Comment by JRI98** [comment]
No longer reproduces

**Comment by lukewilliamboswell** [comment]
Fixed by #6832
